File: w03/d05/hoa_newton/calculator/lib/calculator.rb
class Calculator 

	def add(a, b)
		# if a > 0 && b > 0
			return a + b
		# end	
	end

	def subtract(x, y)
		x - y
	end

	def sum(arr)
		arr.length == 0? 0 : arr.reduce(:+)
		arr.reduce(0, :+)
	end

	def multiply(*num)
		return num.reduce(:*)
	end

	def factorial(num)
		# if num == 0 || num == 1
		# 	return 1
		# else
		# 	(1..num).reduce(:*)
		# end
		num == 0? 1 : (1..num).reduce(:*)
	end

	def power(x, y)
		#x**y
		if y == 0 
			return 1
		else
			x*power(x, y-1)
		end
	end

end

File: w06/d05/francis_castillo/functions/calculator.js
//////Subtract//////
var subtract = function(a, b) {
  return a - b;
};
console.log(subtract(9,3));
/////ADD/////
function add(a, b) {
  return a + b;
}
//////Times//////
var times = function(a, b) {
  newArr = [];
  for (var i = 0; i < b; i++) {
    newArr.push(a);
  }
  return newArr.reduce(add);
};
console.log(times(9,3));
//////Divide////
var divide = function(a, b) {
  var count = 0;
  var newArr = [a,b];
  for (var i = 0; i < b; i++) {
    if (newArr[0] >= newArr[1]) {
      newArr[0] = subtract(newArr[0], newArr[1]);
      count++;
    }
  }
  return count;
};
console.log("dividing");
console.log(divide(9,3));
console.log(divide(25,6));
console.log(divide(25,5));
console.log(divide(5,22));

//////////////////////////////////////
///Factorial/////////////////////////

function factorial(num) {
  var newNumArr = Array.apply( 1, { length: (num + 1)}).map(Number.call, Number);
  newNumArr.shift();
  return newNumArr.reduce(times);
}
console.log(factorial(5));
console.log(factorial(3));

//////////////////////////////////////
/////////Power///////////////////////

function power(a, b) {
  newArr = [];
  if (b === 0) {
    return 1;
  } else {
      for (var i = 0; i < b; i++) {
        newArr.push(a);
      }
  return newArr.reduce(times);
  }
}

console.log(power(3,3));
console.log(power(4,2));
console.log(power(3,1));
console.log(power(3,0));
